release/rancher/rancher.go

```diff
@@ -838,3 +880,47 @@ const checkRancherRCDepsTemplate = `{{- define "componentsFile" -}}
 * {{ .Content }} ({{ .File }}, line {{ .Line }})
 {{- end}}
 {{ end }}`
+
+const updateDashboardReferencesScript = `#!/bin/bash
```

Comment:
Can you comment this explaining what this logic does?

Reply:
Yep, I also thought that it was missing some comments explaining a few of those lines. 

Added a few comments explaining most of the steps in this script.

Reply:
Was this a script taken from somewhere else or you wrote it? If it's ECM code, please update this to be POSIX complaint and use `/bin/sh`. Run this through shellcheck to catch any kind of issues.

Reply:
This looks a lot like the k3s scripts

Reply:
Ran through `shellcheck` there was a couple of warnings, and I fixed those.